docs/concepts/create-read-update-delete.md

```diff
@@ -0,0 +1,127 @@
+# Create, Read, Update, Delete
+
+GalaChain provides a set of utility functions that build on Hyperledger Fabric's state management to provide an improved developer experience. These functions handle validation, serialization, and caching to ensure atomicity guarantees for each transaction.
+
+## Core Operations
+
+### Creating and Updating Objects
+
+The `putChainObject` function handles both creation and updates of chain objects. It validates the object, serializes it, and queues it for writing to the World State.
+
+```typescript
+// Define a chain object
+class GameMove extends ChainObject {
+  @ChainKey(0)
+  gameId: string;
+
+  @ChainKey(1)
+  moveNumber: number;
+
+  piece: string;
+  from: string;
+  to: string;
+  player: string;
+}
+
+// Create and store a new move
+const move = new GameMove({
+  gameId: "game_123",
+  moveNumber: 1,
+  piece: "P",
+  from: "e2",
+  to: "e4",
+  player: "white"
+});
+
+await putChainObject(ctx, move);
+```
+
+### Reading Objects
+
+#### Single Object Lookup
+
+Use `getObjectByKey` to fetch a specific object using its composite key:
+
+```typescript
+// Fetch a specific move
+const moveKey = GameMove.getCompositeKeyFromParts(GameMove.INDEX_KEY, ["game_123", 1]);
+const move = await getObjectByKey(ctx, GameMove, moveKey);
+console.log(move.piece, move.from, move.to); // "P" "e2" "e4"
+```
+
+#### Query Multiple Objects
+
+Use `getObjectsByPartialCompositeKey` to query objects using partial keys:
+
+```typescript
+// Get all moves for a specific game
+const moves = await getObjectsByPartialCompositeKey(
+  ctx,
+  "GameMove",
+  ["game_123"],
+  GameMove
+);
+
+// For large result sets, use pagination
+const { results: paginatedMoves, metadata } = await getObjectsByPartialCompositeKeyWithPagination(
+  ctx,
+  "GameMove",
+  ["game_123"],
+  GameMove,
+  undefined, // bookmark
+  10 // limit
+);
+```
+
+### Deleting Objects
+
+Use `deleteChainObject` to remove an object from the World State:
+
+```typescript
+// Delete a specific move
+const moveKey = GameMove.getCompositeKeyFromParts(GameMove.INDEX_KEY, ["game_123", 1]);
+await deleteChainObject(ctx, moveKey);
+```
+
+## The GalaChainStubCache
+
+GalaChain implements a transactional cache through the `GalaChainStub` class. This cache ensures:
+
+1. **Read Consistency**: Once an object is read within a transaction, subsequent reads return the same value unless explicitly modified.
+
+2. **Write Atomicity**: All changes are queued and only written to the World State when the transaction successfully completes.
+
+3. **Performance**: Frequently accessed objects are cached to reduce ledger reads.
```

Comment:
maybe worth adding something like:

This behavior is different than World State reads in Hyperledger Fabric, when you always get the value that is saved on chain, and in-transaction changes are not visible. (poor words and grammar probably, but you get the idea 😉 )